docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+feature.
+
+##### 4.2.2.1. Implementation
+The `Task` class contains a member `tag` of String type.
+
+The figure below shows how the tag operation works:
+
+![Sequence Diagram for Tag Timetable command](/diagrams/class/jpeg/SequenceDiagram_TagTimetable.jpg)
+
+There are 3 crucial processes during the tag operation. When the user enters `tag 1 /tCS2113T` into the command window while using the application:
+
+**Input Parsing**
+1. `CliUserInterface` receives the "tag 1 /tCS2113T" input by the user and calls the `executeCommand` method.
+2. Method `executeCommand` constructs the `InputParser` class and calls `InputParser#getCommandFromInput` to pass the
+input to `InputParser`.
+3. `InputParser#getCommandFromInput`calls `InputParser#parseTagDescription` to separate the index from the tag.
+4. If the application is in the timetable mode, `InputParser#getCommandFromInput` then parses the index and constructs
+the `TagCommandTimetableMode` class.
+5. `InputParser#getCommandFromInput` returns `TagCommandTimetableMode` back to `CliUserInterface`
+
+**Command Execution**
+1. `CliUserInterface#executeCommand` calls `TagCommandTimetableMode#execute` to execute the command.
+2. In `TagCommandTimetableMode#execute()`, the `CliMessages` class, which prints any outputs to the user, is constructed.
+3. `Tasklist#getTask` returns the `Task` that is specified by the index.
+4. `Task#setTag` changes the `tag` member of `Task` to the tag input by the user.
+5. If the tag is successful, `CliMessages#printTagTaskMessage` displays the message to the user.
+
+**Storage**
+1. `CliUserInterface#executeCommand` finally calls `TagCommandTimetableMode#isTriggerAutoSave` method to check whether a change
+has been made.
+2. If the method returns `True`, `CliUserInterface#executeCommand`calls `saveState` method to save the
+current list.
+3. The tag operation ends.
+
+##### 4.2.2.2. Design Considerations
+This section describes some of the considerations involved when designing the tag feature.
+
+###### Aspect: How to store the tags
+- **Alternative 1 (current choice):** Store as a private `String` member in every task
+    - Pros: It is easy to access for print operations.
+    - Cons: It is unoptimized in terms of complexity for search operations, which requires more work for scaling of the
+    application.
+- **Alternative 2:** Store as a Hash Table with the key as the tag and value as `Task`
+    - Pros: It has a better time complexity for search operations since this data structure is more optimized (O(1) can
+    be achieved).
+    - Cons: It is hard to retrieve the tag for a specific `Task` due to the structure of the key-value pair.
+
+#### 4.2.3. List Feature
+
+##### 4.2.3.1. Implementation
+
+The following sequence diagram shows how the list operation works:
+
+![Sequence Diagram for List command](/diagrams/class/jpeg/SequenceDiagram_List.jpg)
```

Comment:
Not sure if it's a hard rule, but I think the labels for the method calls should be above the lines pointing to the right, not in the middle of the line? e.g. InputParser()

Reply:
<img width="308" alt="Screenshot 2020-10-28 at 12 42 26 PM" src="https://user-images.githubusercontent.com/46882774/97391847-20cc3f80-191b-11eb-8c17-bb93b048b1fc.png">

Reply:
Also, another suggestion: perhaps space out the alt block? All the content within looks very crammed together and hard to read.

Reply:
<img width="443" alt="Screenshot 2020-10-28 at 12 45 14 PM" src="https://user-images.githubusercontent.com/46882774/97391968-786aab00-191b-11eb-9463-bcd99b55ceba.png">

Reply:
Noted, thank you.